service/history/task/task.go

```diff
@@ -288,20 +288,14 @@ func (t *taskImpl) HandleErr(err error) (retErr error) {
 		err = nil
 	}
 
-	// target domain not active error, we should retry the task
-	// so that a cross-cluster task can be created.
-	if err == errTargetDomainNotActive {
-		t.scope.IncCounter(metrics.TaskTargetNotActiveCounterPerDomain)
-		t.logger.Error("Dropping 'domain-not-active' error as non-retriable", tag.Error(err))
-		return nil
```

Comment:
this is the problem I think, it's just dropping a task on the floor which is a problem in a race

Reply:
if the domain is failed over to the other region, the tasks are not able to start child workflow in this region because this region is active. I think even if you retry the task, you'll still get the same error

Reply:
that makes sense @Shaddoll. in that case should we also remove the time elapsed check all together?

Reply:
The logs show that the error was emitted from the *newly* active region, in which case I expect it to converge towards being correct on retry. 

(for context in the failover, I was failing up-cm-dca from PHX *back* to its original location in DCA at 12:43PST, the errors were emitted from *dca*)

Reply:
synced offline and now I understand that there might be a race condition with failover and domain cache update in different history hosts.
Approved, since the risk of the change is very low.

Reply:
Got it. this is to avoid dropping the task on the newly active side 👍